downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

```diff
@@ -42,6 +42,57 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](
 
 ### Improvements                             
 
+#### Create immutable record value using record type with mutable default values
+
+With these improvements, enable the usage of mutable defaults by wrapping them with `cloneReadOnly` when creating an immutable record value with a record type.
+
+```ballerina
+type Temp record {|
+    any[] initalValues = [10, 20, 30];
+|};
+
+function value() {
+    // Now results in compile-time error.
+    Temp & readonly _ = {};
+}
+```
```

Comment:
````suggestion
```ballerina
import ballerina/io;

type Student record {|
    int id;
    string name;
    // The inherent type of the default value expression is `int[]`.
    int[] moduleCodes = [1001, 2001, 3010];
    // The inherent type of the default value expression is `any[]`.
    any[] config = getStudentConfig();
|};

function createEmployee(int id, string name, readonly & string[] config) {
    // No longer panics at runtime, since an immutable value is set
    // for the `moduleCodes` field.
    Student & readonly s1 = {id, name, config};
    io:println(s1.moduleCodes is readonly & int[]); // true
}   

isolated function getStudentConfig() returns any[] {
    return [];
}
```

If the default value does not belong to `value:Cloneable`, and therefore, an immutable value cannot be created by calling `value:cloneReadOnly`, the compiler requires specifying a value for such a field (i.e., the default will not be used).

```ballerina
function createEmployee(int id, string name) {
    // Results in a compile-time error now since there is no default
    // value that can be used for `config`.
    Student & readonly s1 = {id, name};
}   
```
````